src/impl_methods.rs

```diff
@@ -1048,18 +1119,36 @@ impl<A, S, D> ArrayBase<S, D> where S: Data<Elem=A>, D: Dimension
     /// Return a mutable pointer to the first element in the array.
     #[inline(always)]
     pub fn as_mut_ptr(&mut self) -> *mut A
-        where S: DataMut
+        where S: DataRawMut
     {
-        self.ensure_unique(); // for RcArray
+        self.try_ensure_unique(); // for RcArray
         self.ptr
     }
 
+    /// Return a pointer array.
+    #[inline(always)]
```

Comment:
Can be inline or no attribute, same below

Reply:
Fixed.